include/fuzzuf/cli/fuzzer/afl/build_afl_fuzzer_from_args.hpp

```diff
@@ -155,20 +159,57 @@ std::unique_ptr<TFuzzer> BuildAFLFuzzerFromArgs(
 
     using fuzzuf::algorithm::afl::option::GetDefaultOutfile;
     using fuzzuf::algorithm::afl::option::GetMapSize;
+    using fuzzuf::cli::ExecutorKind;
+
+    std::shared_ptr<TExecutor> executor;
+    switch (global_options.executor) {
+    case ExecutorKind::NATIVE: {
+        auto nle = std::make_shared<NativeLinuxExecutor>(
+                            setting->argv,
+                            setting->exec_timelimit_ms,
+                            setting->exec_memlimit,
+                            setting->forksrv,
+                            setting->out_dir / GetDefaultOutfile<AFLTag>(),
+                            GetMapSize<AFLTag>(), // afl_shm_size
+                            0 // bb_shm_size
+                        );
+        executor = std::make_shared<TExecutor>(std::move(nle));
+        break;
+    }
 
-    // Create NativeLinuxExecutor
-    auto nle = std::make_shared<NativeLinuxExecutor>(
-                        setting->argv,
-                        setting->exec_timelimit_ms,
-                        setting->exec_memlimit,
-                        setting->forksrv,
-                        setting->out_dir / GetDefaultOutfile<AFLTag>(),
-                        GetMapSize<AFLTag>(), // afl_shm_size
-                                           0  //  bb_shm_size
-                    );
+    case ExecutorKind::QEMU: {
+        // NOTE: Assuming GetMapSize<AFLTag>() == QEMUExecutor::QEMU_SHM_SIZE
+        auto qe = std::make_shared<QEMUExecutor>(
+                            global_options.exec_path.value(),
+                            setting->argv,
+                            setting->exec_timelimit_ms,
+                            setting->exec_memlimit,
+                            setting->forksrv,
+                            setting->out_dir / GetDefaultOutfile<AFLTag>()
+                        );
+        executor = std::make_shared<TExecutor>(std::move(qe));
+        break;
+    }
 
-    // TODO: support more types of executors
-    auto executor = std::make_shared<TExecutor>(std::move(nle));
+#ifdef __aarch64__
+    case ExecutorKind::CORESIGHT: {
+        auto cse = std::make_shared<CoreSightExecutor>(
+                            global_options.exec_path.value(),
+                            setting->argv,
+                            setting->exec_timelimit_ms,
+                            setting->exec_memlimit,
+                            setting->forksrv,
+                            setting->out_dir / GetDefaultOutfile<AFLTag>(),
+                            GetMapSize<AFLTag>() // afl_shm_size
+                        );
+        executor = std::make_shared<TExecutor>(std::move(cse));
+        break;
+    }
+#endif
+
+    default:
+        EXIT("Unsupport executor: '%s'", global_options.executor.c_str());
```

Comment:
サポートしていないExecutorが指定されたらエラー出して終了して偉い

Reply:
I use `EXIT()` macro here because `ERROR()` is used when `errno` is set and can be used with `perror()` (https://github.com/retrage/fuzzuf/blob/feature/2022-03-07-dyn-executor-cli/include/fuzzuf/logger/logger.hpp#L235). In this case, the error is not a system error, so it is not proper to use `ERROR()` macro here. It should use `EXIT()` macro as you can see the comment (https://github.com/retrage/fuzzuf/blob/feature/2022-03-07-dyn-executor-cli/include/fuzzuf/logger/logger.hpp#L209).
However, this macro is misused everywhere in the fuzzuf code. I think the macro names `ERROR()` and `EXIT()` is somehow misleading, so we should rename them in the future.

Reply:
(Additional follow; No action required) The CLI policy was to throw an exception when an error occurred (see https://github.com/fuzzuf/fuzzuf/blob/master/cli/cli_entry_point.cpp#L48 ). However, this was not communicated and agreed upon by the entire fuzzuf developers, so the use of ERROR() is acceptable. We need to get rid of the ERROR() macro that rejects the exception handler in the first place.